[C++] Add Protocol and websocket documentation (#174)
kuzzleio/documentation-V2: src/sdk-reference/cpp/1/protocol/add-listener/index.md

---
layout: sdk.html.hbs
title: addListener

> description: Add a listener to an event

Reviewer: ```suggestion
description: Adds a listener to an event
```

---

# addListener

Adds a listener to an event.

> ## Signature

Reviewer: Wrong documentation template.

This should be:

```cpp

## Arguments

`<function signature>`

<arguments table>

[optional: ### <argument>
... further explanations]
```

This must be fixed everywhere.

Reviewer: On outdated documentation, it's like this. See our debriefing here: https://kaliop.slack.com/archives/GBJ9VRA01/p1541422858016200

The new format has been applied since then, and a global pass will be performed to standardize articles written before these decisions.

SDK Examples:
https://docs-v2.kuzzle.io/sdk-reference/js/6/collection/create/
https://docs-v2.kuzzle.io/sdk-reference/js/6/ms/georadiusbymember/

API: all documentation pages follow the new format

Author: Hum ok, great... 
I will do this in the last commit, after fixing the other things.

```cpp
virtual void addListener(kuzzleio::Event event, kuzzleio::EventListener* listener) = 0;
```

## Arguments

| Argument   | Type                      | Description
| ---------- |------------------------------------------------------------------------------------------------------ | -------- |

> | `event`    | <pre>kuzzleio::Event</pre>           | An enum representing the listener [event]({{ site_base_path }}sdk-reference/essentials/event-handling)

Reviewer: broken link

```suggestion
| `event`    | <pre>kuzzleio::Event</pre>           | An enum representing the listener [event]({{ site_base_path }}sdk-reference/cpp/1/events)
```

| `listener` | <pre>kuzzleio::EventListener*</pre> | A pointer to a c++11 lambda

### **event**

One of the following event:

```cpp
CONNECTED
DISCARDED
DISCONNECTED
LOGIN_ATTEMPT
NETWORK_ERROR
OFFLINE_QUEUE_POP
OFFLINE_QUEUE_PUSH
QUERY_ERROR
RECONNECTED
JWT_EXPIRED
ERROR
```

### **listener**

> A c++11 lambda which take a `const std::string`

Reviewer: Not sure this suggestion is correct, it seems that a part of the sentence is missing.

```suggestion
A c++11 lambda which takes a `const std::string` argument
```
